src/platform/qt/scripting/AutorunScriptModel.cpp

```diff
@@ -5,22 +5,59 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 #include "scripting/AutorunScriptModel.h"
 
-#include "ConfigController.h"
-#include "LogController.h"
+#include <QtDebug>
+#include <iostream>
+
+QDataStream& operator<<(QDataStream& stream, const QGBA::AutorunScriptModel::ScriptInfo& object) {
+	stream << QGBA::AutorunScriptModel::ScriptInfo::VERSION;
+	stream << object.filename.toUtf8();
+	stream << object.active;
+	return stream;
+}
+
+QDataStream& operator>>(QDataStream& stream, QGBA::AutorunScriptModel::ScriptInfo& object) {
+	uint16_t version = 0;
+	stream >> version;
+	if (version == 1) {
+		QByteArray filename;
+		stream >> filename;
+		object.filename = QString::fromUtf8(filename);
+	} else {
+		qCritical() << QGBA::AutorunScriptModel::tr("Could not load autorun script settings: unknown script info format %1").arg(version);
```

Comment:
It occurs to me to mention that this pops a message box.

Reply:
You should probably change it to qWarning

Reply:
You think so? It felt appropriate to give a popup to inform the user that the configuration that they had previously saved had gotten blown away.

Reply:
That only makes sense if this only shows up once--which, unless we follow up by immediately wiping the corrupted information, won't necessarily be the case.

Reply:
It is the case, though. When the model signals that it's been updated after loading the deserialized data, it triggers a save.

Reply:
Ah. I guess that's fine then.